packages/rabbitmq/src/TxSubmitWorker.ts

```diff
@@ -0,0 +1,273 @@
+   */
+  #connection?: Connection;
+
+  /**
+   * Flag to exit from the infinite loop in case of sequential tx handlig
+   */
+  #continueForever = false;
+
+  /**
+   * The RabbitMQ consumerTag
+   */
+  #consumerTag?: string;
+
+  /**
+   * The function to call to resolve the start method exit Promise
+   */
+  #exitResolver?: () => void;
+
+  /**
+   * The logger
+   */
+  #logger: Logger;
+
+  /**
+   * Worker configuration: it was configured to process multiple tx in parallel
+   */
+  #parallel?: boolean;
+
+  /**
+   * The RabbitMQ polling cycle in milliseconds.
+   */
+  #pollingCycle: number;
+
+  /**
+   * Worker configuration: the RabbitMQ connection URL
+   */
+  #rabbitmqURL: URL;
+
+  /**
+   * The internal worker status
+   */
+  #status: 'connected' | 'connecting' | 'error' | 'idle' = 'idle';
+
+  /**
+   * The TxSubmitProvider instance
+   */
+  #txSubmitProvider: TxSubmitProvider;
+
+  /**
+   * @param {TxSubmitWorkerOptions} options The configuration options
+   */
+  constructor(options: TxSubmitWorkerOptions) {
+    const { logger, parallel, pollingCycle, rabbitmqURL, txSubmitProvider } = options;
+
+    this.#logger = logger || dummyLogger;
+    this.#parallel = parallel;
+    this.#pollingCycle = pollingCycle || 500;
+    this.#rabbitmqURL = rabbitmqURL;
+    this.#txSubmitProvider = txSubmitProvider;
+  }
+
+  /**
+   * Get the status of the worker
+   *
+   * @returns the status of the worker
+   */
+  getStatus() {
+    return this.#status;
+  }
+
+  /**
+   * Starts the worker
+   */
+  start() {
+    return new Promise<void>(async (resolve, reject) => {
+      const closeHandler = async (async: boolean, err: unknown) => {
+        if (err) {
+          if (async) this.#logger.error('TxSubmitWorker: Asyncronous error');
+          this.logError(err);
+          this.#exitResolver = undefined;
+          this.#status = 'error';
+          await this.stop();
+          reject(err);
+        }
+      };
+
+      try {
+        this.#logger.info('TxSubmitTxSubmitWorker init: checking tx submission provider health status');
+
+        const { ok } = await this.#txSubmitProvider.healthCheck();
+
+        if (!ok) throw new ProviderError(ProviderFailure.Unhealthy);
+
+        this.#logger.info('TxSubmitWorker init: opening RabbitMQ connection');
+        this.#exitResolver = resolve;
+        this.#status = 'connecting';
+        this.#connection = await connect(this.#rabbitmqURL.toString());
+        this.#connection.on('close', (error) => closeHandler(true, error));
+
+        this.#logger.info('TxSubmitWorker init: opening RabbitMQ channel');
+        this.#channel = await this.#connection.createChannel();
+        this.#channel.on('close', (error) => closeHandler(true, error));
+
+        this.#logger.info('TxSubmitWorker init: ensuring RabbitMQ queue');
+        await this.#channel.assertQueue(TX_SUBMISSION_QUEUE);
+        this.#logger.info('TxSubmitWorker: init completed');
+
+        if (this.#parallel) {
```

Comment:
Is this really processing jobs in "parallel"? The logic I'm reading in the `submitTx` function is to `ack` after successful submission, which would be serial. In review, I think we need to look at actually _buffering_ the jobs being consumed, rather than "parallelisation", as all we're trying to do is reduce latency when there's a large number of txs to process. 

Take a look at:
https://www.rabbitmq.com/consumer-prefetch.html
https://amqp-node.github.io/amqplib/channel_api.html#channel_prefetch

Reply:
> is it 10 ok?

Let's default to 3, and we can adjust after perf testing.

> Isn't actually RabbitMQ the one which plays the role of the buffer?

Adding a buffer to the consumer just reduces the impact of the latency between the broker and worker when processing a long queue.